libs/web3/src/lib/web3-provider.tsx

```diff
@@ -1,14 +1,23 @@
 import type { Web3ReactHooks } from '@web3-react/core';
 import { Web3ReactProvider } from '@web3-react/core';
 import type { Connector } from '@web3-react/types';
+import { useMemo } from 'react';
 
 interface Web3ProviderProps {
   children: JSX.Element | JSX.Element[];
   connectors: [Connector, Web3ReactHooks][];
 }
 
 export const Web3Provider = ({ children, connectors }: Web3ProviderProps) => {
+  const key = useMemo(
```

Comment:
suggestion (non blocking): can you add a comment as to why we need this key? Its unclear why this key is required